### ISSUE TITLE
nz-tree-select: This component causes memory leak due creating a tons of dom notes

### ISSUE THREAD

**Issue by Yorer** [NG-ZORRO/ng-zorro-antd#6785]
### Steps to reproduce
- Create 2 routes to be able to navigate between them without reloading the page
- Add nz-tree-select to your component on a route
- Open Chome debugger (F12 or Ctrl Shift I) and open "Performance monitor" in it.
- Make sure "JS heap size" and "Dom note" selected
- Navigate between routes and check heap size

Heap size increasing like hell which cause a mega memory leak in the entire application.

### What is expected?
After garbage collection heap size should be decreased to it's original size.

### What is actually happening?
It is increasing like hell until it takes all of my memory.

| Environment | Info |
|---|---|
| ng-zorro-antd | 11.4.2 |
| Browser | Chrome |

**Comment by Greegko** [comment]
I tried to reproduce this but without success: https://stackblitz.com/edit/angular-nz-tree-leak

Are you sure your component is destroyed ?

**Comment by Yorer** [comment]
Of course it is destroyed I also change route and go back to the route where a tree select available. Very interesting! Yes, by your example it is actually working fine... I don't really understand maybe I missed something in my code then. When I call your nz-tree select in your code my dom increasing so much, but just only if that plugin added, if I remove it everything is fine. I've tried to replicate it here in stackblitz with the same object im getting as response, and I was also unable to replicate. Also I've tried promise as well on stackblitz without any problem.. I'm so confused now. So I'm closing this issue now for further checking. Thanks Greeko for your time. I have no idea which causing memory leak then. I was pretty sure this causing it but now I don't know.
